solr/core/src/java/org/apache/solr/core/SolrCores.java

```diff
@@ -490,21 +492,19 @@ public CoreDescriptor getCoreDescriptor(String coreName) {
   }
 
   /**
-   * Get the CoreDescriptors for every SolrCore managed here
-   * @return a List of CoreDescriptors
+   * Get the CoreDescriptors for every {@link SolrCore} managed here (permanent and transient, loaded and unloaded).
+   *
+   * @return An unordered list copy. This list can be modified by the caller (e.g. sorted).
```

Comment:
Collectors.toList says that there is no guarantees on the mutability.  So remove the docs about the mutability guarantee of what we return because we can't guarantee that.
Or, simply use ArrayList and call addAll.  I prefer this; use of Streams here doesn't appear to add value.

Reply:
I'll use ArrayList.